server/src/test/java/org/elasticsearch/cluster/metadata/NodesShutdownMetadataTests.java

```diff
@@ -121,6 +122,36 @@ public void testSigtermIsRemoveInOlderVersions() throws IOException {
         assertThat(new SingleNodeShutdownMetadata(out.bytes().streamInput()).getType(), equalTo(SingleNodeShutdownMetadata.Type.SIGTERM));
     }
 
+    public void testIsNodeMarkedForRemoval() {
+        SingleNodeShutdownMetadata.Type type;
+        SingleNodeShutdownMetadata.Builder builder = SingleNodeShutdownMetadata.builder()
+            .setNodeId("myid")
+            .setReason("myReason")
+            .setStartedAtMillis(0L);
+        switch (type = randomFrom(SingleNodeShutdownMetadata.Type.values())) {
```

Comment:
might be worth adding the case where the node ID is not there at all with a random shutdown type.